pallets/collator-selection/src/lib.rs

```diff
@@ -375,7 +402,14 @@ pub mod pallet {
 				index,
 				<frame_system::Pallet<T>>::block_number(),
 			);
-			Some(Self::assemble_collators())
+			let candidates_len = Self::candidates().len();
+			let result = Some(Self::assemble_collators());
+			let removed = candidates_len - result.clone().unwrap_or_default().len();
+			frame_system::Pallet::<T>::register_extra_weight_unchecked(
+				T::WeightInfo::new_session(candidates_len as u32, removed as u32),
```

Comment:
Can you explain how the weight of the new session is now being calculated?

Reply:
yup, I take the worse case scenerio which is the max size collators with all of them being removed. I test for that in the benchmark (c... and r... T::MaxCandidates). Then at the end of the call I pass through how big the candidates vec was and how many were removed